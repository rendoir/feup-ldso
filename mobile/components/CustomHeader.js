import React from 'react';
import {
    Image,
    StatusBar
} from 'react-native';
import {
    View,
    Text
} from 'native-base';
import SwitchToggle from 'react-native-switch-toggle';
import '../global.js';

export default class CustomHeader extends React.Component {
    constructor(props) {
        super(props);

        this.toggleLanguage = this.toggleLanguage.bind(this);
    }

    render() {
        return (
            <View style={{ height: '6%' }}>
                <StatusBar hidden />
                <View style={{ backgroundColor: '#002040', height: '100%', flexDirection: 'row' }}>
                    <View style={{ flex: 2, height: '100%' }}>
                        <Image source={require('../assets/images/original.png')}
                            style={{
                                width: '80%',
                                height: '100%',
                                resizeMode: 'contain',
                                marginHorizontal: '1%',
                                marginVertical: '0%'
                            }}
                        />
                    </View>
                    <View>
                        <SwitchToggle
                            buttonText={this.props.language === 'EN' ? 'EN' : ''}
                            backTextRight={"PT"}
                            backTextLeft={"EN"}
                            switchOn={this.props.language === 'PT'}
                            onPress={this.toggleLanguage}
                            type={1}
                            buttonStyle={{
                                alignItems: 'center',
                                justifyContent: 'center',
                                position: 'absolute'
                            }}
                            rightContainerStyle={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}
                            leftContainerStyle={{ flex: 1, alignItems: 'center', justifyContent: 'flex-start' }}

                            buttonTextStyle={{ fontSize: 9 }}
                            textRightStyle={{ fontSize: 9 }}
                            textLeftStyle={{ fontSize: 9 }}
                            backgroundColorOn='#fff'
                            backgroundColorOff='#fff'
                            circleColorOff='#e5e1e0'
                            circleColorOn='#e5e1e0'
                            className="change-language"
                            accessibilityLabel="Toggle language"
                        />
                    </View>
                    <View style={{ flex: 2, alignSelf: 'center' }}>
                        <Text style={{
                            color: 'white', textAlign: 'right', paddingRight: '1%', marginHorizontal: '1%'
                        }}>{global.userName}</Text>
                    </View>
                </View>
            </View>
        );
    }

    toggleLanguage() {
        this.props.toggleLanguage();
    }
}
