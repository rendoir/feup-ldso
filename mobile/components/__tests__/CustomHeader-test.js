import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CustomHeader from '../CustomHeader';

it('renders correctly', () => {
    const tree = renderer.create(<CustomHeader/>).toJSON();

    expect(tree).toMatchSnapshot();
});
