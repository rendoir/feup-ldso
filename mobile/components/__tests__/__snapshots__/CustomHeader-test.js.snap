// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Header renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "height": "6%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#002040",
          "flexDirection": "row",
          "height": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 2,
            "height": "100%",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": "100%",
            "marginHorizontal": "1%",
            "marginVertical": "0%",
            "resizeMode": "contain",
            "width": "80%",
          }
        }
      />
    </View>
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderRadius": 18,
              "flexDirection": "row",
              "height": 36,
              "padding": 3,
              "width": 72,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "flex-start",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 9,
                }
              }
            >
              EN
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(229, 225, 224, 1)",
                "borderRadius": 15,
                "height": 30,
                "justifyContent": "center",
                "position": "absolute",
                "transform": Array [
                  Object {
                    "translateX": 36,
                  },
                ],
                "width": 30,
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 9,
                  }
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 9,
                }
              }
            >
              PT
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignSelf": "center",
            "flex": 2,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "white",
              "marginHorizontal": "1%",
              "paddingRight": "1%",
              "textAlign": "right",
            },
          ]
        }
        uppercase={false}
      />
    </View>
  </View>
</View>
`;
